### PR TITLE
log fix for Issue #931

### LINKED ISSUE #931: Wrong log message for received json
**Describe the bug**
The `logJson` function is used by `receivingMQTT` and `pub`, to log the json which is send and received.
However, the log message text is always `Received json : %s`.
This means also when sending json, in the logs it shows as received json.

**To Reproduce**
1. Open the serial monitor
2. Wait for `SYStoMQTT`
3. See wrong log message

**Expected behavior**
The log message should either indicate the correct action (sending, receiving) or exclude this information from the log.

**Screenshots**
```
N: Subject: /SYStoMQTT
N: Received json : {"uptime":1803,"version":"v0.9.6","freemem":206120,...}
T: jsonPublishing
```

**Environment:**
* OpenMQTTGateway version: development

**Additional context**
I use log level `LOG_LEVEL_VERBOSE`.

### PR DESCRIPTION
**Pull request by Odyno**
A trivial PR, useful for understanding logs. It fix #931

### CODE REVIEW

File: main/main.ino [hunk 1]
```diff
@@ -453,14 +453,18 @@ void pubMQTT(String topic, unsigned long payload) {
   client.publish((char*)topic.c_str(), val);
 }
 
-void logJson(JsonObject& jsondata) {
+void logJson(String comment, JsonObject& jsondata) {
```
Comment (1technophile) [inline main/main.ino]: Could you use a char * instead? it would avoid the c_.str() below

Comment (Odyno) [reply main/main.ino]: ok!

**Comment by 1technophile** [comment]
Thanks, could you apply lint format and  rebase from development branch please

**Comment by 1technophile** [comment]
@odyno seems like we are missing 4 bytes for the UNO, could you try to reduce the length of the log messages so as to fit in it, and rebase please.

**Comment by Odyno** [comment]
@1technophile  Done if it is still valid

**Comment by Odyno** [comment]
@1technophile  done with lint, reduced the dimension of code for uno-rf

File: main/main.ino [hunk 2]
```diff
@@ -281,6 +281,17 @@ bool to_bool(String const& s) { // thanks Chris Jester-Young from stackoverflow
   return s != "0";
 }
 
+template <class T>
```
Comment (Legion2) [inline main/main.ino:284]: I think this generic parameter causes issues

**Comment by Odyno** [comment]
closed in favour of #1093

**Pull request closed without merging**